rosbag2/src/rosbag2/sequential_reader.cpp

```diff
@@ -41,11 +45,12 @@ void
 SequentialReader::open(
   const StorageOptions & storage_options, const ConverterOptions & converter_options)
 {
+  metadata_ = metadata_io_->read_metadata(storage_options.uri);
   storage_ = storage_factory_->open_read_only(storage_options.uri, storage_options.storage_id);
```

Comment:
So how does this help? We still pass the same `uri` to the open call. How did you test this latest change?

Reply:
I just noticed that `test_rosbag2_play_end_to_end` is disabled

Reply:
Ok I changed `open_read_only` to accept `metadata_.relative_file_paths[0]`

Steps to test:
1. run `ros2 bag record -a` in first terminal
2. run `ros2 run demo_nodes_py talker` in another terminal
3. ctrl+c second terminal
4. ctrl+c first terminal
5. run `ros2 run listener` in first terminal
6. run `ros2 bag play rosbag2_2019_11_14-16_19_43/`
7. kill all terminals

Output:
```
[INFO] [listener]: I heard: [Hello World: 0]
[INFO] [listener]: I heard: [Hello World: 1]
```